core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -78,7 +78,8 @@ private FunctionResolver date() {
   private FunctionResolver dayOfMonth() {
     return define(DAYOFMONTH.getName(),
         impl(nullMissingHandling(DateTimeFunction::exprDayOfMonth),
-            INTEGER, DATE)
+            INTEGER, DATE),
+        impl(nullMissingHandling(DateTimeFunction::exprDayOfMonth), INTEGER, STRING)
```

Comment:
could you confirm this? dayofmonth doesn't accept DATE? e.g. dayofmonth(CURDATE()) should work.

Reply:
It accepts date, this line is to addd string as argument as well